### PR TITLE
[C-1554] Add phantom connect + sign message flow

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

- Finalizes phantom connect + sign message flow
- Adds mobile-specific wallet-connect slice to manage in-app flow
- Removes component based effects in favor of sagas
- Prepares for refactoring common token-dashboard sagas to break up flow into chunks that can be platform agnostic
- Adds in-progress wallets to wallets-list

https://user-images.githubusercontent.com/8230000/204918868-961f05f2-4148-468c-a50b-42e06b4250e3.mov

### CODE REVIEW

**Comment by raymondjacobson** [comment]
This is awesome!!!!

File: packages/common/src/store/pages/token-dashboard/selectors.ts [hunk 1]
```diff
@@ -47,3 +47,5 @@ export const getCanRecipientReceiveWAudio = (state: CommonState) => {
   }
   return 'false'
 }
+export const getConfirmingWallet = (state: CommonState) =>
```
Comment (raymondjacobson) [inline packages/common/src/store/pages/token-dashboard/selectors.ts:50]: merge conflict on this, oops :(

File: packages/mobile/src/components/navigation-container/NavigationContainer.tsx [hunk 2]
```diff
@@ -141,8 +143,24 @@ const NavigationContainer = (props: NavigationContainerProps) => {
       // Add leading slash if it is missing
       if (path[0] !== '/') path = `/${path}`
 
-      // Remove the app-redirect prefix if present
+      const walletConnectPath = /^\/(wallet-connect)/
```
Comment (raymondjacobson) [inline packages/mobile/src/components/navigation-container/NavigationContainer.tsx:146]: cool

**Review by raymondjacobson — APPROVED** [review]
This is all super nice

File: packages/mobile/src/screens/wallet-connect/WalletConnectScreen.tsx [hunk 3]
```diff
@@ -51,6 +52,17 @@ export const WalletConnectScreen = () => {
   const styles = useStyles()
   const navigation = useNavigation<WalletConnectParamList>()
   const connector = useWalletConnect()
+  const dispatch = useDispatch()
+  const { params } = useRoute<WalletConnectRoute<'Wallets'>>()
+
+  useEffect(() => {
+    if (!params) return
+    if (params.path === 'wallet-connect') {
+      dispatch(connectNewWallet(params))
```
[the remark this answers is not in the document]
Comment (dylanjeffers) [reply packages/mobile/src/screens/wallet-connect/WalletConnectScreen.tsx:61]: Yeah I liked how this turned out

**Pull request merged**